File: Javascript/Lecture01/v8.cpp
#include<iostream>
using namespace std;

int main(){
    string code;
    getline(cin,code);
    // Now print the element
    int open  = code.find('(');
    int close = code.rfind(')');
    string inside = code.substr(open+1,close-open-1);

    if(inside[0]=='"'){
        cout<<inside.substr(1,inside.size()-2)<<endl;
    }
    else{
        int index = 0, a = 0, b = 0;
        while((inside[index]!='+')&&(inside[index]!='-')&&(inside[index]!='*')&&(inside[index]!='/')){
            a = a*10 + (inside[index]-'0');
            index++;
        }
        char op = inside[index];
        index++;

        while(index<inside.size()){
            b = b*10 + (inside[index]-'0');
            index++;
        }
        if(op=='+'){
            cout<<a+b<<endl;
        }
        else if(op=='-'){
            cout<<a-b<<endl;
        }
        else if(op=='*'){
            cout<<a*b<<endl;
        }
        else{
            cout<<a/b<<endl;
        }
    }

    return 0;

}
 
// console.log("hello  Chandan");

File: 02Javascript/Lecture01/v8.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut code = String::new();
    if stdin.lock().read_line(&mut code).is_err() {
        eprintln!("Failed to read input");
        return;
    }
    let Some(open) = code.find('(') else {
        eprintln!("Missing opening parenthesis");
        return;
    };
    let Some(close) = code.rfind(')') else {
        eprintln!("Missing closing parenthesis");
        return;
    };
    if close <= open {
        eprintln!("Invalid expression");
        return;
    }
    let inside = code[open + 1..close].trim();

    if inside.starts_with('"') && inside.ends_with('"') && inside.len() >= 2 {
        println!("{}", &inside[1..inside.len() - 1]);
        return;
    }
    let Some((operator_index, operator)) = inside
        .char_indices()
        .find(|(_, character)| matches!(character, '+' | '-' | '*' | '/'))
    else {
        eprintln!("Missing arithmetic operator");
        return;
    };
    let left = inside[..operator_index].trim().parse::<i64>();
    let right = inside[operator_index + 1..].trim().parse::<i64>();

    let (Ok(a), Ok(b)) = (left, right) else {
        eprintln!("Invalid integer expression");
        return;
    };
    let result = match operator {
        '+' => Some(a + b),
        '-' => Some(a - b),
        '*' => Some(a * b),
        '/' if b != 0 => Some(a / b),
        '/' => {
            eprintln!("Cannot divide by zero");
            None
        }
        _ => None,
    };
    if let Some(value) = result {
        println!("{value}");
    }
}
